core/os/flock/flock.go

```diff
@@ -173,15 +176,33 @@ func (m *mutex) unlock() error {
 
 // ReleaseAllLocks releases all the FLocks by removing all the underlying files.
 func ReleaseAllLocks() error {
-	info, err := os.Stat(dir)
+	dirInfo, err := os.Stat(dir)
 	if os.IsNotExist(err) {
 		return nil
 	}
 	if err != nil {
 		return err
 	}
-	if info.IsDir() {
-		return os.RemoveAll(dir)
+	if !dirInfo.IsDir() {
+		return fmt.Errorf("%v is not a directory", dir)
+	}
+	err = os.RemoveAll(dir)
```

Comment:
Überdoobernit:
In the case of short lines like this, it's fine to combine the assignment and if:

```go
if err := os.RemoveAll(dir); err == nil {
    return nil
}
```